### PR TITLE
Spreadsheet substations modification

### PR DESCRIPTION
**Pull request by ayolab**

### CODE REVIEW

File: src/components/spreadsheet/utils/equipement-table-popup-editors.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        const intl = useIntl();
+        const [open, setOpen] = useState(true);
+        const [localRowData, setRowData] = useState(() => {
+            const data = rowData;
+            if (!data?.properties) {
+                return [];
+            }
+            const keys = Object.keys(data.properties);
+            return keys.map((key, index) => {
+                return { id: index, key: key, value: data.properties[key] };
```
Comment (AAJELLAL) [inline src/components/spreadsheet/utils/equipement-table-popup-editors.js:66]: May be it's better to extract this to a single method then use it here to initiate the state.

File: src/components/spreadsheet/utils/equipement-table-popup-editors.js
Comment (AAJELLAL) [inline src/components/spreadsheet/utils/equipement-table-popup-editors.js:1]: As a general rule new files should be in TS

**Review by AAJELLAL — CHANGES_REQUESTED** [review]

**Review by souissimai — COMMENTED** [review]
At the moment of validation, the country code is shown on the country column, it seems a strange behavior

**Comment by jamal-khey** [comment]
> src/components/spreadsheet/utils/equipement-table-popup-editors.js

this is fixed

File: src/components/spreadsheet/utils/equipement-table-popup-editors.js [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                            >
+                                <Table stickyHeader size="small">
+                                    <PropertiesEditorHeader
+                                        darkTheme={
+                                            theme.palette.mode === 'dark'
+                                        }
+                                        handleAddRow={handleAddRow}
+                                    />
+                                    <TableBody>
+                                        {localRowData?.map((row, index) => {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/spreadsheet/utils/equipement-table-popup-editors.js:176]: in my case key is not always unique and the user can change it. but the index is more stable

File: src/components/spreadsheet/utils/equipement-table-popup-editors.tsx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            const newData = [...localRowData];
+            newData[index].key = value;
+            setRowData(newData);
+        };
+
+        const handleValueChange = (index: number, value: any) => {
+            const newData = [...localRowData];
+            newData[index].value = value;
+            setRowData(newData);
+        };
```
Comment (AAJELLAL) [inline src/components/spreadsheet/utils/equipement-table-popup-editors.tsx:144]: You can refactor this, it's practically the same code.

**Review by AAJELLAL — CHANGES_REQUESTED** [review]

**Review by AAJELLAL — APPROVED** [review]

**Pull request merged**